### PR TITLE
CloudWatch: Log group variable should get all log groups

### LINKED ISSUE #50532: Template variable query function for listing CloudWatch log groups is limited to 50 items
**What happened**:

The template variable query function for listing CloudWatch log groups is limited to 50 items. For instance, If we filter by `/aws/lambda`  in a region with more than 200 lambda functions we only get the 50 first (in alphabetical order):

image

**What you expected to happen**:

Return all available log groups.

**How to reproduce it (as minimally and precisely as possible)**:

Use the [`Log Groups`](https://github.com/grafana/grafana/pull/50100) query type and preview values.

**Anything else we need to know?**:

Initial implementation in https://github.com/grafana/grafana/pull/50100

**Environment**:
- Grafana version: v8.5.5 (d32ae18909)
- Data source type & version: CloudWatch
- OS Grafana is installed on: CentOS 7
- User OS & Browser: GNU/Linux & Google Chrome
- Grafana plugins: n/a
- Others: n/a

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #50532

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/tsdb/cloudwatch/log_actions.go [hunk 1]
```diff
@@ -235,6 +237,42 @@ func (e *cloudWatchExecutor) handleDescribeLogGroups(ctx context.Context,
...
+	}
+
+	var response *cloudwatchlogs.DescribeLogGroupsOutput
+	var err error
+	logGroupNames := []*string{}
+	for {
+		response, err = logsClient.DescribeLogGroupsWithContext(ctx, &cloudwatchlogs.DescribeLogGroupsInput{
+			LogGroupNamePrefix: namePrefix,
+			NextToken:          nextToken,
+			Limit:              aws.Int64(10),
```
Comment (sunker) [inline pkg/tsdb/cloudwatch/log_actions.go:255]: Any reason this is 10 and not 50 (default)?

File: public/app/plugins/datasource/cloudwatch/datasource.ts [hunk 2]
```diff
@@ -474,6 +474,13 @@ export class CloudWatchDatasource
     return logGroupNames;
   }
 
+  async describeAllLogGroups(params: DescribeLogGroupsRequest): Promise<string[]> {
+    const dataFrames = await lastValueFrom(this.makeLogActionRequest('DescribeAllLogGroups', [params]));
+
+    const logGroupNames = dataFrames[0]?.fields[0]?.values.toArray() ?? [];
```
Comment (sunker) [inline public/app/plugins/datasource/cloudwatch/datasource.ts:480]: Unrelated to this PR of course, but at some point we should move the DescribeLogsGroups request to the resource handler so we don't have to do this mad parsing from data frames.

**Review by sunker — APPROVED** [review]
Is there a reason the capability to return all pages wasn't added to the existing DescribeLogsGroup func? Could have been a new param. This approach is fine, I'm just curious. 

Previously, grafana instances have crashed due to out of memory when iterating over thousands of pages in the ListMetrics api. To guard up against that we added a [limit](https://github.com/grafana/grafana/blob/main/pkg/tsdb/cloudwatch/metric_find_query.go#L508) that could be overridden in the config file. Since this is a new template variable query, I don't expect this to cause major problems in our production environments. We can handle that later if we run into the same problem.

File: pkg/tsdb/cloudwatch/log_actions.go (hunk 1 above)
Comment (iwysiu) [reply pkg/tsdb/cloudwatch/log_actions.go]: no, I was using that to test and forgot to commit the cleanup changes 😞

**Review by kevinwcyu — APPROVED** [review]

**Pull request merged**